hack/versions-management/github_auth.sh

```
@@ -0,0 +1,42 @@
+if ! command -v jwt &> /dev/null; then
+  echo "Downloading jwt-cli tool..."
+  mkdir -p "$HOME/.local/bin"
+  curl -L https://github.com/mike-engel/jwt-cli/releases/download/6.2.0/jwt-linux.tar.gz -o "jwt-linux.tar.gz"
+  tar -xzf "jwt-linux.tar.gz"
+  mv "jwt" "$HOME/.local/bin/"
+  chmod +x "$HOME/.local/bin/jwt"
+  export PATH="$HOME/.local/bin:$PATH"
+  echo "jwt-cli tool installed successfully"
+fi
+
+JWT=$(jwt encode --alg=RS256 --secret=@$GITHUB_APP_PRIVATE_KEY_PATH '{"iss":"'$GITHUB_APP_ID'", "exp":'$(date -d "+600 seconds" +%s)'}')
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)